Assets/Scripts/ColumnMovement.cs

```diff
@@ -10,30 +11,38 @@ public class ColumnMovement : MonoBehaviour
 
     private const float Speed = -1.5f;
 
+    private Vector2 _columnMotionVector;
+
     private void Start()
     {
-        _rigidbody.velocity = new Vector2(Speed, 0);
+        _columnMotionVector = new Vector2(Speed, 0);
     }
 
     private void Update()
     {
-        if (_gameController.IsGameOver || _pauseController.IsPause)
-        {
-            _rigidbody.velocity = Vector2.zero;
-        }
-        else
-        {
-            _rigidbody.velocity = new Vector2(Speed, 0);
-        }
+        MoveTheObject(_gameController, _pauseController, _rigidbody,
+            _columnMotionVector);
+        
+        
     }
 
-    public void SendAGameControllerReference(GameController gameController)
+    public void Inject(GameController gameController, PauseController pauseController)
     {
         _gameController = gameController;
+        _pauseController = pauseController;
     }
-
-    public void SendAPauseControllerReference(PauseController pauseController)
+    
+    public void MoveTheObject(GameController gameController, PauseController pauseController, Rigidbody2D rb2D, Vector2 vector2)
```

Comment:
you already have 
 [SerializeField] private Rigidbody2D _rigidbody; 
 why do you pass it as a parameter?